flink-formats/flink-protobuf/src/main/java/org/apache/flink/formats/protobuf/PbFormatUtils.java

```diff
@@ -0,0 +1,166 @@
+public class PbFormatUtils {
+
+    /**
+     * protobuf code has a bug that, f_abc_7d will be convert to fAbc7d in {@code
+     * com.google.protobuf.Descriptors.FileDescriptor.getJsonName()}, but actually we need fAbc7D.
+     */
+    public static String fieldNameToJsonName(String name) {
+        final int length = name.length();
+        StringBuilder result = new StringBuilder(length);
+        boolean isNextUpperCase = false;
+        for (int i = 0; i < length; i++) {
+            char ch = name.charAt(i);
+            if (ch == '_') {
+                isNextUpperCase = true;
+            } else if (isNextUpperCase) {
+                if ('a' <= ch && ch <= 'z') {
+                    ch = (char) (ch - 'a' + 'A');
+                    isNextUpperCase = false;
+                }
+                result.append(ch);
+            } else {
+                result.append(ch);
+            }
+        }
+        return result.toString();
+    }
+
+    private static String getJavaPackageFromProtoFile(Descriptors.Descriptor descriptor) {
+        boolean hasJavaPackage = descriptor.getFile().getOptions().hasJavaPackage();
+        if (hasJavaPackage) {
+            String javaPackage = descriptor.getFile().getOptions().getJavaPackage();
+            if (StringUtils.isBlank(javaPackage)) {
+                throw new FlinkRuntimeException("java_package cannot be blank string");
+            }
+            return javaPackage;
+        } else {
+            String packageName = descriptor.getFile().getPackage();
+            if (StringUtils.isBlank(packageName)) {
+                throw new FlinkRuntimeException("package and java_package cannot both be empty");
+            }
+            return packageName;
+        }
+    }
+
+    public static String getFullJavaName(Descriptors.Descriptor descriptor) {
+        String javaPackageName = getJavaPackageFromProtoFile(descriptor);
+        if (descriptor.getFile().getOptions().getJavaMultipleFiles()) {
+            // multiple_files=true
+            if (null != descriptor.getContainingType()) {
+                // nested type
+                String parentJavaFullName = getFullJavaName(descriptor.getContainingType());
+                return parentJavaFullName + "." + descriptor.getName();
+            } else {
+                // top level message
+                return javaPackageName + "." + descriptor.getName();
+            }
+        } else {
+            // multiple_files=false
+            if (null != descriptor.getContainingType()) {
+                // nested type
+                String parentJavaFullName = getFullJavaName(descriptor.getContainingType());
+                return parentJavaFullName + "." + descriptor.getName();
+            } else {
+                // top level message
+                if (!descriptor.getFile().getOptions().hasJavaOuterClassname()) {
+                    // user do not define outer class name in proto file
+                    return javaPackageName
```

Comment:
@maosuhan 
1.I find some bugs. When i don't declare **java_outer_classname=XXX** or **java_multiple_files=true** option in file .proto(proto2)  this will be generated x.y.z.xxxOuterClass.MyPBClass. It is incorrect.

generated code: public static RowData decode(data.LogOuterClass.Log message){
my pb code data.Data.Log

2.How to identify inner class?

Reply:
Here's my pb
```protobuf
syntax = "proto2";
package data;

message ValueType {
    optional int32 int_type =1;
    optional int64 long_type = 2;
    optional float float_type = 3;
    optional string string_type =4;

}

message Log {
    required int64 log_timestamp =1;
    optional string ip = 2;
    optional group Field  = 3 {
     repeated group Map = 1 {
            required string key = 1;
            optional ValueType value = 2;
        }
    }
}


```

Reply:
@MyLanPangzi Thanks for your findings. Yes, it is a bug. I found that if the proto file name is test.proto and the content is as below, the outermost class name will be TestOuterClass because the message name is the same with file name.
```
syntax = "proto2";
message Test {
....
}
```

It is a little tricky to get java full class name from the `Descriptors.Descriptor`, so I use a simpler way that I extract the outer prefix name from `protobuf.message-class-name`.
```
    public static String getOuterProtoPrefix(String name) {
        name = name.replace('$', '.');
        int index = name.lastIndexOf('.');
        if (index != -1) {
            // include dot
            return name.substring(0, index + 1);
        } else {
            return "";
        }
    }
``` 
I have fixed that issue and added additional unit tests.